extension/js/common/core/crypto/smime/smime-key.ts

```diff
@@ -29,8 +29,8 @@ export class SmimeKey {
       expiration: SmimeKey.dateToNumber(certificate.validity.notAfter),
       fullyDecrypted: false,
       fullyEncrypted: false,
-      isPublic: true,
-      isPrivate: true,
+      isPublic: !!certificate.publicKey,
+      isPrivate: !!certificate.privateKey,
```

Comment:
The way we use these fields, currently I think, is either/or.

So either isPublic is true, or isPrivate, not both.

That means if there is both public and private key included, I think we'd set isPublic to false and isPrivate to true. (for consistency with pgp)